File: core/src/test/java/com/ataulm/chunks/ChunksFixtures.java
package com.ataulm.chunks;

import static com.ataulm.chunks.ItemsFixtures.items;

public final class ChunksFixtures {

    private ChunkDate todaysDate;
    private Items today;
    private Items tomorrow;
    private Items sometime;

    public static ChunksFixtures aChunks() {
        return new ChunksFixtures(
                ChunkDate.create(0),
                items().get(),
                items().get(),
                items().get()
        );
    }

    private ChunksFixtures(ChunkDate todaysDate, Items today, Items tomorrow, Items sometime) {
        this.todaysDate = todaysDate;
        this.today = today;
        this.tomorrow = tomorrow;
        this.sometime = sometime;
    }

    public ChunksFixtures withTodaysDate(ChunkDate todaysDate) {
        this.todaysDate = todaysDate;
        return this;
    }

    public ChunksFixtures withToday(Items today) {
        this.today = today;
        return this;
    }

    public ChunksFixtures withTomorrow(Items tomorrow) {
        this.tomorrow = tomorrow;
        return this;
    }

    public ChunksFixtures withSometime(Items sometime) {
        this.sometime = sometime;
        return this;
    }

    public Chunks get() {
        return Chunks.create(todaysDate, today, tomorrow, sometime);
    }

}

File: core/src/test/java/com/ataulm/chunks/ChunksFixtures.kt
package com.ataulm.chunks

import com.ataulm.chunks.ItemsFixtures.items

class ChunksFixtures private constructor(
        private var todaysDate: ChunkDate,
        private var today: Items,
        private var tomorrow: Items,
        private var sometime: Items
) {

    fun withTodaysDate(todaysDate: ChunkDate): ChunksFixtures {
        this.todaysDate = todaysDate
        return this
    }

    fun withToday(today: Items): ChunksFixtures {
        this.today = today
        return this
    }

    fun withTomorrow(tomorrow: Items): ChunksFixtures {
        this.tomorrow = tomorrow
        return this
    }

    fun withSometime(sometime: Items): ChunksFixtures {
        this.sometime = sometime
        return this
    }

    fun get(): Chunks {
        return Chunks.create(todaysDate, today, tomorrow, sometime)
    }

    companion object {

        fun aChunks(): ChunksFixtures {
            return ChunksFixtures(
                    ChunkDate.create(0),
                    items().get(),
                    items().get(),
                    items().get()
            )
        }
    }

}
